imap_processing/ialirt/tests/unit/test_decom_ialirt.py

```diff
@@ -0,0 +1,197 @@
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from imap_processing import imap_module_directory
+from imap_processing.decom import decom_packets
+from imap_processing.ialirt.l0.decom_ialirt import generate_xarray
+
+IALIRT_PACKET_LENGTH = 1464
+
+
+@pytest.fixture(scope="session")
+def xtce_ialirt_path():
+    """Returns the xtce auxiliary directory."""
+    return imap_module_directory / "ialirt" / "packet_definitions" / "ialirt.xml"
+
+
+@pytest.fixture(scope="session")
+def binary_packet_path():
+    """
+    Creates a binary file from the text packet data, which is more representative
+    of the actual operational environment. The binary file is deleted after the
+    test session.
+    """
+    packet_path = (
+        imap_module_directory
+        / "ialirt"
+        / "tests"
+        / "test_data"
+        / "l0"
+        / "IALiRT Raw Packet Telemetry.txt"
+    )
+
+    with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+        binary_file_path = Path(tmp_file.name)
+
+    with open(packet_path) as text_file, open(binary_file_path, "wb") as binary_file:
+        for line in text_file:
+            if not line.startswith("#"):
+                # Split the line by semicolons
+                # Discard the first value since it is only a counter
+                hex_values = line.strip().split(";")[1:]
+                # Convert hex to binary
+                binary_data = bytearray.fromhex("".join(hex_values))
+                binary_file.write(binary_data)
+
+    yield binary_file_path
+
+    binary_file_path.unlink()
+
+
+@pytest.fixture(scope="session")
+def empty_binary_packet_path():
+    """
+    Creates an empty binary file, representative of a scenario
+    in which no packet data is available.
+    """
+    with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+        binary_file_path = Path(tmp_file.name)
+
+    yield binary_file_path
+
+    binary_file_path.unlink()
```

Comment:
I'd remove this entire block and put the relevant tmp_path usage inside the test below since you're only using it once it isn't really a fixture.
```suggestion
```